packages/appkit/src/client/appkit.ts

```diff
@@ -520,63 +519,90 @@ export class AppKit extends AppKitBaseClient {
   }
 
   protected override async injectModalUi() {
-    if (!isInitialized && CoreHelperUtil.isClient()) {
-      const features = { ...CoreConstantsUtil.DEFAULT_FEATURES, ...this.options.features }
-
-      // Selectively import views based on feature flags
-      const featureImportPromises = []
-      if (features) {
-        // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
-        const usingEmbeddedWallet = features.email || (features.socials && features.socials.length)
-        if (usingEmbeddedWallet) {
-          featureImportPromises.push(import('@reown/appkit-scaffold-ui/embedded-wallet'))
-        }
+    // Skip entirely in non-browser environments - ensures proper tree-shaking during build
+    if (
+      !CoreHelperUtil.isClient() ||
+      typeof window === 'undefined' ||
+      typeof document === 'undefined'
```

Comment:
Maybe we can instead update `isClient()` function as:

```js
isClient() {
    return typeof window !== 'undefined' || typeof document !== 'undefined'
  }
```

Reply:
lol this was not intended for this pr 
nut I'll include it anyways as it's a bit nicer for bundlers